openlane/steps/step.py

```diff
@@ -1121,3 +1138,71 @@ def list(Self) -> List[str]:
             return [cls.id for cls in Self.__registry.values()]
 
     factory = StepFactory
+
+
+class CompositeStep(Step):
+    """
+    A step composed of other steps, run sequentially. The steps are intended
+    to run as a unit within a flow and cannot be run separately.
+
+    Composite steps are currently considered an internal object that is not
+    ready to be part of the API. The API may change at any time for any reason.
+    """
+
+    Steps: List[Type[Step]] = []
+
+    def __init_subclass__(Self):
+        super().__init_subclass__()
+        available_inputs = set()
+
+        input_set: Set[DesignFormat] = set()
+        output_set: Set[DesignFormat] = set()
+        config_var_dict: Dict[str, Variable] = {}
+        for step in Self.Steps:
+            for input in step.inputs:
+                if input in available_inputs:
+                    continue
+                else:
+                    input_set.add(input)
+                    available_inputs.add(input)
+            for output in step.outputs:
+                available_inputs.add(input)
+                output_set.add(output)
```

Comment:
This means that the outputs of the composite step will include the outputs of the intermediate steps. I believe a composite step should always declare the list of final outputs.

Reply:
I made it so if you explicitly declare the outputs, it will use your list, but if you don't it will generate them.